src/pymovements/gaze/experiment.py

```diff
@@ -93,42 +95,48 @@ def __init__(
             distance_cm: float | None = None,
             origin: str | None = 'upper left',
             sampling_rate: float | None = None,
+            *,
+            screen: Screen | None = None,
             eyetracker: EyeTracker | None = None,
     ):
-        self.screen = Screen(
-            width_px=screen_width_px,
-            height_px=screen_height_px,
-            width_cm=screen_width_cm,
-            height_cm=screen_height_cm,
-            distance_cm=distance_cm,
-            origin=origin,
-        )
-
+        checks.check_is_mutual_exclusive(screen_width_px=screen_width_px, screen=screen)
+        checks.check_is_mutual_exclusive(screen_height_px=screen_height_px, screen=screen)
+        checks.check_is_mutual_exclusive(screen_width_cm=screen_width_cm, screen=screen)
+        checks.check_is_mutual_exclusive(screen_height_cm=screen_height_cm, screen=screen)
+        checks.check_is_mutual_exclusive(distance_cm=distance_cm, screen=screen)
         checks.check_is_mutual_exclusive(sampling_rate=sampling_rate, eyetracker=eyetracker)
 
+        # the origin default value needs special care.
+        if origin != 'upper left' or (screen is not None and screen.origin != origin):
```

Comment:
this needs to be improved, otherwise a user needs to make sure that the explicit `origin` is equal to `screen.origin`

Reply:
I rewrote this to that:

https://github.com/aeye-lab/pymovements/blob/a96571b4345ec2c9507dc3d781560ae37f330680/src/pymovements/gaze/experiment.py#L109-L112

This should be fine now.